### PR TITLE
WIP Issue#134. Add Contact Entity to application.

### PR DESCRIPTION
**Pull request by alexorlenko**
dev
## ZenHub

* [Main ZenHub ticket](https://app.zenhub.com/workspaces/home-project-5f7b77ff8db73a001cb17009)

## Code reviewers

- [ ] @PoLytvy 
- [ ] @Dandria1802 
- [ ] @Abhai2016 
- [ ] @OlTym 
- [ ] @Romanov-Niko 

### Second Level Review

- [ ] @kh0ma 

## Summary of issue

Add Contact Entity to application.

## Summary of change

Delete some tables into liquibase changelog.

## Testing approach

-

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Comment by kh0ma** [comment]
@alexorlenko 
Why do we need an additional pr/issue for adding it? Can we just walk it through #91 ?

File: home-data-migration/src/main/resources/db/changelog/0.0.1/Issue134.xml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                      http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+
+    <changeSet id="Issue134.001" author="alexorlenko">
+        <createTable tableName="contact">
```
Comment (OlTym) [inline home-data-migration/src/main/resources/db/changelog/0.0.1/Issue134.xml:9]: may be add  <preConditions onFail="MARK_RAN">
            <not>
                <tableExists tableName="contact"/>
            </not>
        </preConditions>

File: home-api-tests/pom.xml [hunk 2]
```diff
@@ -249,5 +249,10 @@
             <artifactId>commons-text</artifactId>
             <version>${commons-text-version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.glassfish.jersey.core</groupId>
+            <artifactId>jersey-common</artifactId>
+            <scope>test</scope>
+        </dependency>
```
Comment (Abhai2016) [inline home-api-tests/pom.xml:256]: Why do we need it?

File: home-application/src/main/java/com/softserveinc/ita/homeproject/application/api/UserApiImpl.java [hunk 3]
```diff
@@ -93,13 +106,15 @@ public Response getAllUsers(@Min(1) Integer pageNumber,
...
+                                String contactPhone,
+                                String contactEmail) {
 
         Map<QueryParamEnum, String> filterMap = new HashMap<>();
 
         filterMap.put(UserQueryParamEnum.ID, id);
         filterMap.put(UserQueryParamEnum.EMAIL, email);
-        filterMap.put(UserQueryParamEnum.CONTACT, contact);
+        filterMap.put(UserQueryParamEnum.CONTACT_PHONE, contactPhone);
+        filterMap.put(UserQueryParamEnum.CONTACT_EMAIL, contactEmail);
```
Comment (Abhai2016) [inline home-application/src/main/java/com/softserveinc/ita/homeproject/application/api/UserApiImpl.java]: Not sure that it's gonna work.

File: home-application/src/main/java/com/softserveinc/ita/homeproject/application/mapper/config/ReadPhoneContactHomeMappingConfig.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+@Component
+@RequiredArgsConstructor
+public class ReadPhoneContactHomeMappingConfig implements HomeMappingConfig<PhoneContactDto, ReadContact> {
+
+    @Lazy
+    private final HomeMapper homeMapper;
+
+    @Override
+    public void addMappings(TypeMap<PhoneContactDto, ReadContact> typeMap) {
+        typeMap.setProvider(request -> homeMapper.convert(request.getSource(), ReadPhoneContact.class));
```
Comment (Abhai2016) [inline home-application/src/main/java/com/softserveinc/ita/homeproject/application/mapper/config/ReadPhoneContactHomeMappingConfig.java:21]: I'm curious if we can map without a lot of configs.

File: home-data/src/main/java/com/softserveinc/ita/homeproject/homedata/entity/Email.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+import lombok.*;
+
+import javax.persistence.Column;
+import javax.persistence.DiscriminatorValue;
+import javax.persistence.Entity;
+
+@Getter
+@Setter
+@Builder
```
Comment (Abhai2016) [inline home-data/src/main/java/com/softserveinc/ita/homeproject/homedata/entity/Email.java]: We don't need it anymore.

File: home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/service/impl/ContactServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    @Transactional
+    @Override
+    public ContactDto updateContact(Long id, ContactDto updateContactDto) {
+        Optional<Contact> contactOptional = contactRepository.findById(id);
+        if (contactOptional.isPresent()) {
+            Contact contact = contactOptional.get();
+            if (contact instanceof Phone) {
+                Phone phone = (Phone) contact;
```
Comment (Abhai2016) [inline home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/service/impl/ContactServiceImpl.java]: I guess we can do it in different methods.

**Pull request closed without merging**